test/test_overrides.py

```diff
@@ -878,5 +878,72 @@ def f(a):
 
         self.assertEqual(f(A()), -1)
 
+class TestIndexing(TestCase):
+    """ Regression tests for gh-46277 """
+    def test_getitem(self):
+        class A:
+            @classmethod
+            def __torch_function__(cls, func, types, args, kwargs=None):
+                return -1
+
+        t = torch.tensor([5])
+        self.assertEqual(t[A()], -1)
+        self.assertEqual(t, torch.tensor([5]))
+
+    def test_getitem_subclass(self):
+        class A(torch.Tensor):
+            @classmethod
+            def __torch_function__(cls, func, types, args, kwargs=None):
+                return -1
+
+        t = torch.tensor([5])
+        self.assertEqual(t[A()], -1)
+        self.assertEqual(t[A(), A()], -1)
```

Comment:
A more convincing test here would be if the first index was a normal number (to show that we are testing all indices)

Reply:
Done; including all the other locations.